Framework/Algorithms/src/CalculateCountRate.cpp

```diff
@@ -0,0 +1,719 @@
+      "The energy mode for 'RangeUnits' conversion mode (default: elastic)");
+
+  // Used logs group
+  std::string used_logs_mode("Used normalization logs");
+  declareProperty(
+      "NormalizeTheRate", true,
+      "Usually you want to normalize counting rate to some "
+      "rate related to the source beam intensity. Change this to "
+      "'false' if appropriate time series log is broken || not attached to "
+      "the input workspace.");
+  declareProperty("UseLogDerivative", false, "If the normalization log gives "
+                                             "cumulative counting, derivative "
+                                             "of this log is necessary to get "
+                                             "correct normalization values.");
+  declareProperty(
+      "NormalizationLogName", "proton_charge",
+      "The name of the log, used in the counting rate normalization. ");
+  declareProperty(
+      "UseNormLogGranularity", true,
+      "If true, the calculated log will have the normalization log "
+      "accuracy; If false, the 'NumTimeSteps' in the visualization "
+      "workspace below will be used for the target log granularity too.");
+  setPropertyGroup("NormalizeTheRate", used_logs_mode);
+  setPropertyGroup("UseLogDerivative", used_logs_mode);
+  setPropertyGroup("NormalizationLogName", used_logs_mode);
+  setPropertyGroup("UseNormLogGranularity", used_logs_mode);
+
+  declareProperty(
+      "CountRateLogName", "block_count_rate",
+      boost::make_shared<Kernel::MandatoryValidator<std::string>>(),
+      "The name of the processed time series log with count rate to be added"
+      " to the source workspace");
+  // visualization group
+  std::string spur_vis_mode("Spurion visualization");
+  declareProperty(
+      Kernel::make_unique<API::WorkspaceProperty<>>(
+          "VisualizationWs", "", Kernel::Direction::Output,
+          API::PropertyMode::Optional),
+      "Optional name to build 2D matrix workspace for spurion visualization. "
+      "If name is provided, a 2D workspace with this name will be created "
+      "containing data to visualize counting rate as function of time in the "
+      "ranges XMin-XMax");
+
+  auto mustBeReasonable = boost::make_shared<Kernel::BoundedValidator<int>>();
+  mustBeReasonable->setLower(3);
+  declareProperty(
+      Kernel::make_unique<Kernel::PropertyWithValue<int>>(
+          "NumTimeSteps", 200, mustBeReasonable, Kernel::Direction::Input),
+      "Number of time steps (time accuracy) the visualization workspace has. "
+      "Also number of steps in 'CountRateLogName' log if "
+      "'UseNormLogGranularity' is set to false. Should be bigger than 3");
+  declareProperty(
+      Kernel::make_unique<Kernel::PropertyWithValue<int>>(
+          "XResolution", 100, mustBeReasonable, Kernel::Direction::Input),
+      "Number of steps (accuracy) of the visualization workspace has along "
+      "X-axis. ");
```

Comment:
Mantid dies, if *XResolution* is given a value-too-large. Would there be a sensible upper limit this property? Or maybe these kinds of deaths should be prevented at some lower lever of the software.